react/week4/my-app/src/user.context.js

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+
+export const UserContext = React.createContext();
+
+class UserProvider extends React.Component {
+  state = {
+    users: [],
+    loading: false,
+    error: null, 
+  };
+
+  updateUserContext = (value) => {
+    this.setState(value)
+  }  
+  
+  render() {
+    return (
+      <UserContext.Provider
+        value={{ ...this.state, updateUserContext: this.updateUserContext }}
+      >
+        {this.props.children}
+      </UserContext.Provider>
+    );
+  }
+}
+ export default UserProvider
```

Comment:
Note that there is no end of line here (`\n`).

In general, we try to keep an end of line character on every line, to indicate that the line ended.

Another advantage of having an end of line is the following: suppose the following code:

```
a = 1[\n]
b = 2
```
When we want to add a new line to this code, 

```
a = 1[\n]
b = 2[\n]
c = 3[\n]
```

we needed to add an end of line in line `b = 2`. This implies that git will tell us that there was one line changed (`b = 2` -> `b = 2[\n]` and one new line `c = 3[\n]`. However, this does not represent well what we were really doing: we were just adding a new line.

Reply:
HI Jorge, thank you so much for your feed back 😊.
yes that make sense of course, thanks for explain that